### PR TITLE
added max_scrape_size to scrape config

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

Related to https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6429

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: lib/flagutil/bytes.go [hunk 1]
```diff
@@ -53,79 +53,77 @@ func (b *Bytes) Set(value string) error {
 		return nil
 	}
 	value = normalizeBytesString(value)
+	N, err := parseBytes(value)
+	if err != nil {
+		return err
+	}
+	b.N = N
+	b.valueString = value
```
Comment (Haleygo) [inline lib/flagutil/bytes.go:61]:
```suggestion
	n, err := parseBytes(value)
	if err != nil {
		return err
	}
	b.N = n
	b.valueString = value
```

**Review by Haleygo — COMMENTED** [review]
`maxScrapeSize.N` -> `c.maxScrapeSize` must be changed in here as well
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/dc3c4e8dfdb4f6633ee2b69e8306fa39ccb1b55e/lib/promscrape/client.go#L163-L166

File: lib/promscrape/scrapework.go [hunk 2]
```diff
@@ -500,6 +504,7 @@ func (sw *scrapeWork) processDataOneShot(scrapeTimestamp, realTimestamp int64, b
 	am := &autoMetrics{
 		up:                        up,
 		scrapeDurationSeconds:     scrapeDurationSeconds,
+		scrapeResponseSize:        float64(len(bodyString)) + float64(unsafe.Sizeof(bodyString)),
```
Comment (zekker6) [inline lib/promscrape/scrapework.go:507]: Could you elaborate why we are including size of `bodyString` here as well?
As far as I can see, the size of response in bytes will be equal to `len(bodyString)` but maybe I'm missing something.

Comment (AndrewChubatiuk) [reply lib/promscrape/scrapework.go]: updated this

File: lib/promscrape/client.go [hunk 3]
```diff
@@ -160,7 +160,7 @@ func (c *client) ReadData(dst *bytesutil.ByteBuffer) error {
 		}
 		return fmt.Errorf("cannot read data from %s: %w", c.scrapeURL, err)
 	}
-	if int64(len(dst.B)) >= maxScrapeSize.N {
+	if int64(len(dst.B)) >= c.maxScrapeSize {
 		maxScrapeSizeExceeded.Inc()
 		return fmt.Errorf("the response from %q exceeds -promscrape.maxScrapeSize=%d; "+
 			"either reduce the response size for the target or increase -promscrape.maxScrapeSize command-line flag value", c.scrapeURL, maxScrapeSize.N)
```
Comment (Haleygo) [inline lib/promscrape/client.go:166]: Or increase `max_scrape_size` under scrape_config.

File: lib/promscrape/scrapework.go [hunk 4]
```diff
@@ -833,6 +839,7 @@ func isAutoMetric(s string) bool {
 func (sw *scrapeWork) addAutoMetrics(am *autoMetrics, wc *writeRequestCtx, timestamp int64) {
 	sw.addAutoTimeseries(wc, "up", float64(am.up), timestamp)
 	sw.addAutoTimeseries(wc, "scrape_duration_seconds", am.scrapeDurationSeconds, timestamp)
+	sw.addAutoTimeseries(wc, "scrape_response_size_bytes", am.scrapeResponseSize, timestamp)
```
Comment (Haleygo) [inline lib/promscrape/scrapework.go:842]: I'm not sure where is this metric will be used, as we already have `vm_promscrape_scrape_response_size_bytes_bucket` to record the response size.

Comment (AndrewChubatiuk) [reply lib/promscrape/scrapework.go:842]: adding `scrape_response_size_bytes` to auto metrics allows to track response size for a specific scrape target and to understand which values should be set for scrape size limits and for which targets

File: lib/promscrape/client.go (hunk 3 above)
Comment (AndrewChubatiuk) [reply lib/promscrape/client.go]: thanks, updated

File: lib/promscrape/scrapework.go (hunk 4 above)
Comment (Haleygo) [reply lib/promscrape/scrapework.go:842]: What about adding this info to the `/targets` page of vmagent ui, like `scrapeDuration` or `samplesScraped`?
image

Comment (AndrewChubatiuk) [reply lib/promscrape/scrapework.go:842]: done

**Review by Haleygo — APPROVED** [review]
LGTM!

**Review by hagen1778 — APPROVED** [review]
Thank you!

**Comment by valyala** [comment]
@AndrewChubatiuk , see the follow-up commit 57000f5105fb360f8b6a04eb13e97f7efed078b5

**Pull request merged**